FoodFriends/app/src/main/java/com/example/foodfriends/models/Restaurant.java

```diff
@@ -87,6 +91,67 @@ public void setAddress(String address){
         put(ADDRESS_KEY, address);
     }
 
+    public String getYelpID(){return getString(YELP_ID_KEY);}
+    public void setYelpId(String yelpID){put(YELP_ID_KEY, yelpID);}
+
+    public static Restaurant fromJson(JSONObject jsonObject) throws JSONException {
+
+        try{
+            String yelp_id = jsonObject.getString("id");
+            ParseQuery<Restaurant> query = ParseQuery.getQuery(Restaurant.class);
+            query.whereEqualTo(YELP_ID_KEY, yelp_id);
+            int num = query.find().size();
+            Log.i(TAG, "NUM EXISTING " + String.valueOf(num));
+            if (num > 0)
+                return null;
+            Log.i(TAG, "MAKING RESTAURANT");
+            Restaurant r = new Restaurant();
+            r.setYelpId(jsonObject.getString("id"));
+            r.setName(jsonObject.getString("name"));
+            r.setImageUrl(jsonObject.getString("image_url"));
+            JSONObject location = jsonObject.getJSONObject("location");
+            r.setCity(location.getString("city"));
+            r.setState(location.getString("state"));
+            r.setZipcode(location.getString("zip_code"));
+            JSONObject coordinates = jsonObject.getJSONObject("coordinates");
+            r.setLatitude(coordinates.getDouble("latitude"));
+            r.setLongitude(coordinates.getDouble("longitude"));
+            r.setPrice("price");
+            String address = "";
+            JSONArray display =  location.getJSONArray("display_address");
+            for(int i = 0; i < display.length(); i++){
```

Comment:
minor: Code formatting looks a bit off. There's a nifty tool in android studio. Code -> Reformat Code,  and it'll handle all the formatting for you :)

Reply:
Thanks! Just reformatted this file, will probably also use this tool for other files